src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -673,6 +679,151 @@ public void testOrderOfCheckExecution() throws Exception {
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException2.java")),
+        };
+        final Map<String, List<String>> expectedViolation = new HashMap<>();
+        expectedViolation.put(getPath("InputTreeWalkerProperFileExtension.java"),
+                new ArrayList<>());
+
+        verify(checker, files, expectedViolation);
+    }
+
+    @Test
+    public void testJavaParseExceptionSeverityDefaultError() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(TreeWalker.class);
+        config.addProperty("skipFileOnJavaParseException", "true");
+        config.addChild(createModuleConfig(NoCodeInFileCheck.class));
+
+        final Checker checker = createChecker(config);
+
+        final File[] files = {
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException.java")),
+            new File(getPath("InputTreeWalkerProperFileExtension.java")),
+            new File(getNonCompilablePath("InputTreeWalkerSkipParsingException2.java")),
+        };
+
+        final Map<String, List<String>> expectedViolation = new HashMap<>();
+
+        expectedViolation.put(getPath("InputTreeWalkerProperFileExtension.java"),
+                new ArrayList<>());
+        expectedViolation.put(getNonCompilablePath("InputTreeWalkerSkipParsingException.java"),
+                List.of("1: Got an exception - IllegalStateException occurred while parsing file "
+                        + getNonCompilablePath("InputTreeWalkerSkipParsingException.java") + "."));
+        expectedViolation.put(getNonCompilablePath("InputTreeWalkerSkipParsingException2.java"),
+                List.of("1: Got an exception - Cannot invoke"
+                        + " \"com.puppycrawl.tools.checkstyle.DetailAstImpl."
+                        + "addPreviousSibling(com.puppycrawl.tools.checkstyle.api.DetailAST)"
+                        + "\" because \"ident\" is null"));
+
+        verify(checker, files, expectedViolation);
+    }
+
+    @Test
+    public void testJavaParseExceptionSeverity() throws Exception {
+        final PackageObjectFactory factory = new PackageObjectFactory(
+                new HashSet<>(), Thread.currentThread().getContextClassLoader());
+        final File testFile = new File(
+                getNonCompilablePath("InputTreeWalkerSkipParsingException.java"));
+
+        for (SeverityLevel severityLevel : SeverityLevel.values()) {
+            final DefaultConfiguration config = createModuleConfig(TreeWalker.class);
+            config.addProperty("skipFileOnJavaParseException", "true");
+            config.addProperty("javaParseExceptionSeverity", severityLevel.getName());
+            config.addChild(createModuleConfig(NoCodeInFileCheck.class));
+
+            final TreeWalker treeWalker = new TreeWalker();
+            treeWalker.setModuleFactory(factory);
+            treeWalker.configure(config);
+
+            final SortedSet<Violation> violations = treeWalker.process(testFile,
```

Comment:
I assume we are calling `TreeWalker` directly, as our normal full run with `Checker` doesn't allow us to see different severity levels?

If so, add a javadoc saying so as we normally want full runs and not direct calls like this.